### PR TITLE
initial otlp tracing support

### PR DESCRIPTION
**Pull request by joshrwolf**
initial otlp implementation, includes:

* default grpc client otlp interceptors
*  helper methods for setting up otlp server interceptors
* replaces deprecated `grpc.WithInsecure` with recommended `grpc.WithTransportCredentials(insecure.NewCredentials())`

### CODE REVIEW

File: pkg/tracing/tracing.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package tracing
```
Comment (mattmoor) [inline pkg/tracing/tracing.go:1]: We should add a license header

File: pkg/tracing/tracing.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		propagation.Baggage{},
+	)
+	otel.SetTextMapPropagator(prp)
+
+	return tp, nil
+}
+
+// NewOTLPExporter returns an OTEL compatible exporter given a grpc endpoint
+func NewOTLPExporter(ctx context.Context, endpoint string, dialOpts ...grpc.DialOption) (*otlptrace.Exporter, error) {
+	conn, err := grpc.DialContext(ctx, endpoint, dialOpts...)
```
Comment (mattmoor) [inline pkg/tracing/tracing.go:35]: I'm wondering whether we should treat `endpoint` here the same way we do with `GRPCOptions` and call that here to get our additional "standard" options, WDYT?

**Review by mattmoor — COMMENTED** [review]
I think it'd be helpful to understand how/where we're going to use this in mono (which is hard to discuss here!), but if you have a PR that pulls this in I could look at to understand the "proper" usage of these tracing methods, it'd be awesome to poke through 🙏

**Comment by k4leung4** [comment]
closing in favor of, https://github.com/chainguard-dev/go-grpc-kit/pull/21, which borrowed from this pr.

**Pull request closed without merging**